woocommerce/woocommerce-gutenberg-products-block
# Update add to cart to useStoreCart hook

Reworked #1984 and I think this also covers the fix in #2093 due to using the apiFetchWithHeaders action.

Fixes #2092

Converts the add to cart functionality which was previously using `__experimentalPersistItemToCollection` to instead use the `useStoreCart` hook.

To support this, a new `cart/add-to-cart` endpoint has been added to the store API. I've reworked some of the routes/schemas to support this and ensure each route returns the correct schema.

**How to test the changes in this Pull Request:**
Copied from #2093:

- Set up a page with All Products block (if needed).
- View the page in an incognto window.
- Add multiple products to cart - should work correctly.
- Optional: also test adding to cart when logged in as a regular customer, and other add to cart flows.

@Aljullu do you get the same issue with this branch?
